src/plot.cpp

```diff
@@ -637,51 +639,80 @@ Plot::Plot(pugi::xml_node plot_node)
   set_mask(plot_node);
 } // End Plot constructor
 
-//==============================================================================
-// POSITION_RGB computes the red/green/blue values for a given plot with the
-// current particle's position
-//==============================================================================
+template<class D>
+D PlotBase::generate_data() const {
```

Comment:
Generally a template definition needs to be in a header. Right now the only place the template is used is in plot.cpp which is why this compiles fine. Are we ok with the assumption that the only place it will ever be used is in plot.cpp?

Reply:
I'm pretty sure it will only be used here, but I can't see any reason not to put it in the header and avoid a problem for someone else in the future.